task/src/task.rs

```diff
@@ -411,21 +411,39 @@ impl<'pin, HSTT: HotShotTaskTypes> ProjectedHST<'pin, HSTT> {
     ) -> Either<Poll<HotShotTaskCompleted>, bool> {
         let event_stream = self.event_stream.take();
         if let Some(mut inner_event_stream) = event_stream {
+            let mut i = 0;
             while let Poll::Ready(maybe_event) = inner_event_stream.as_mut().poll_next(cx) {
+                tracing::error!("YEET I IS: {:?}", i);
                 if let Some(event) = maybe_event {
                     if let Some(handle_event) = self.handle_event {
                         let maybe_state = self.state.take();
                         if let Some(state) = maybe_state {
                             let mut fut = handle_event(event, state);
                             match fut.as_mut().poll(cx) {
                                 Poll::Ready((result, state)) => {
-                                    *self.in_progress_fut = None;
-                                    *self.state = Some(state);
                                     if let Some(completed) = result {
+                                        *self.in_progress_fut = None;
+                                        *self.state = Some(state);
                                         *self.r_val = Some(completed);
                                         let result = self.launch_shutdown_fut(cx);
                                         *self.event_stream = Some(inner_event_stream);
                                         return Left(result);
+                                    } else {
+                                        let mut fut = async move {
+                                            async_yield_now().await;
+                                            (None, state)
+                                        }
+                                        .boxed();
+                                        if let Poll::Ready((_, state)) = fut.as_mut().poll(cx) {
+                                            *self.state = Some(state);
```

Comment:
Comment: explain why it's ok to continue (executor decided no other tasks need to run)